### PR TITLE
Intermediate.ATX.2017.Jordan.Roman

### PR DESCRIPTION
**Pull request by jroman93**
# Checkpoint Rubric

This is the rubric that your instructor will use to grade your checkpoints. Please do not edit. 

## Checkpoint 1: Towers of Hanoi Terminal App
- [X] All tests passed: 55 points
- [ ] Properly commented code: 15 points
0
- [ ] Properly indented code: 15 points
6/15 - missed several
- [X] Demonstrated effective use of JavaScript: 15 points

Total: 76

Specific Comments:
See the github file comments for details

*Comments may also be made on the file itself using github's commenting tools.*

## Checkpoint 2: Towers of Hanoi GUI App
- [X] All tests passed: 55 points
- [X] Properly commented code: 15 points
- [X] Properly indented code: 15 points
- [X] Demonstrated effective use of jQuery: 15 points

Total - 100

Specific Comments:

*Comments may also be made on the file itself using github's commenting tools.*

## Checkpoint 3: Final Project Gist Blog
- [X] Effective use of AJAX: 20 points
- [X] Effective use of loops: 20 points
- [X] Effective use of github API: 20 points
- [X] Style and design: 10 points
- [] Properly commented code: 15 points
0/15 no comments :/
- [X] Properly indented code: 15 points

Specific Comments:

Great work! Very readable. 85/100

*Comments may also be made on the file itself using github's commenting tools.*

### CODE REVIEW

File: apps/03TicTacToe.js [hunk 1]
```diff
@@ -3,55 +3,97 @@
...
-      getPrompt();
+    printBoard();
+    console.log("It's Player " + playerTurn + "'s turn.");
+    rl.question('row: ', (row) => {
+        rl.question('column: ', (column) => {
+            ticTacToe(row, column);
+            getPrompt();
+        });
     });
-  });
```
Comment (zacacollier) [inline apps/03TicTacToe.js:97]: Nice job on this app! You're off to a great start :)

File: apps/03TicTacToe.js [hunk 2]
```diff
@@ -3,55 +3,97 @@
...
-  // Your code here
+if (horizontalWin()) {
+  return true;
+} else if (verticalWin()) {
+  return true;
+} else if (diagonalWin()) {
+  return true;
+} else {
+  return false;
+}
```
Comment (TheFoxWitch) [inline apps/03TicTacToe.js:69]: Good use of else if.

File: apps/04TowersOfHanoi.js [hunk 3]
```diff
@@ -18,25 +18,57 @@ function printStacks() {
...
+function isLegal(startStack, endStack) {
   // Your code here
-
+  var startBlock = stacks[startStack][stacks[startStack].length -1];
+
+  var endBlock =
+  stacks[endStack][stacks[endStack].length -1];
+  // console.log('blocks', startBlock, endBlock);
+  if (stacks[startStack].length === 0) {
+    console.log('You can not do that silly')
```
Comment (tjefferson08) [inline apps/04TowersOfHanoi.js:35]: Nice

File: apps/04TowersOfHanoi.js [hunk 4]
```diff
@@ -18,25 +18,57 @@ function printStacks() {
...
+
+    return false;
+  }
+  if (stacks[endStack].length === 0) {
```
Comment (tjefferson08) [inline apps/04TowersOfHanoi.js:39]: Good early returns! Nice checking of all the scenarios

File: apps/04TowersOfHanoi.js [hunk 5]
```diff
@@ -18,25 +18,57 @@ function printStacks() {
...
+    console.log('You can totally do that')
+    return true;
+  }
+  if (startBlock < endBlock) {
+    console.log ('Good Move!')
+    return true;
+}
+if (startBlock > endBlock) {
```
Comment (tjefferson08) [inline apps/04TowersOfHanoi.js:47]: Hmm what about if they are exactly the same!?!?!?!?

File: apps/08jquery-towers-of-hanoi/script.js [hunk 6]
```diff
@@ -1,5 +1,42 @@
 'use strict';
+var $block = null;
 
 $(document).ready(function() {
   // Put app logic here
+  $('div[data-stack]').on('click', function() {
+    //to do
```
Comment (tjefferson08) [inline apps/08jquery-towers-of-hanoi/script.js:7]: TODO? Looks like you accomplished this task, so make sure you clean up the TODOs when they're done!

File: apps/08jquery-towers-of-hanoi/script.js [hunk 7]
```diff
@@ -1,5 +1,42 @@
...
+    // Only move the block if the value of the data- block attribute is less than the last block of the stack, or if the stack is empty.
+
+    if ($block === null) {
+      //the data stack that was clicked on and you are picking it up
```
Comment (tjefferson08) [inline apps/08jquery-towers-of-hanoi/script.js:11]: This looks like it'd be better as two separate comments

File: apps/08jquery-towers-of-hanoi/script.js [hunk 8]
```diff
@@ -1,5 +1,42 @@
 'use strict';
+var $block = null;
 
 $(document).ready(function() {
   // Put app logic here
+  $('div[data-stack]').on('click', function() {
+    //to do
+    // Only move the block if the value of the data- block attribute is less than the last block of the stack, or if the stack is empty.
+
+    if ($block === null) {
```
Comment (tjefferson08) [inline apps/08jquery-towers-of-hanoi/script.js:10]: Yay ===

File: apps/08jquery-towers-of-hanoi/script.js [hunk 9]
```diff
@@ -1,5 +1,42 @@
...
+      //the data stack that was clicked on and you are picking it up
+      console.log('picking it up')
+      $block = $(this)
+        .children()
+        .last()
+        .detach();
+    } else {
+      //put it back down
```
Comment (tjefferson08) [inline apps/08jquery-towers-of-hanoi/script.js:18]: Good comment

File: apps/08jquery-towers-of-hanoi/script.js [hunk 10]
```diff
@@ -1,5 +1,42 @@
...
+      if ($(this).children().length === 0 || $(this).children().last().data('block') > $block.data('block')) {
+        console.log('putting it down')
+        $(this).append($block);
+        $block = null;
+
+      } else {
+        console.log('no mam')
```
Comment (tjefferson08) [inline apps/08jquery-towers-of-hanoi/script.js:25]: A comment here might help a bit. What does it mean if we get here?

File: apps/08jquery-towers-of-hanoi/script.js [hunk 11]
```diff
@@ -1,5 +1,42 @@
...
+      }
+    }
+    checkForWin();
+  });
+
+  function checkForWin() {
```
Comment (tjefferson08) [inline apps/08jquery-towers-of-hanoi/script.js:31]: great helper function

File: apps/11gist-blog/script.js [hunk 12]
```diff
@@ -1,5 +1,35 @@
+
 'use strict';
 
 $(document).ready(function() {
-  // You code here
+    // You code here
+    $.ajax('https://api.github.com/users/jroman93/gists', {
+        success: function(posts) {
+            posts.forEach(function(post) {
```
Comment (tjefferson08) [inline apps/11gist-blog/script.js:8]: Nice forEach

File: apps/11gist-blog/script.js [hunk 13]
```diff
@@ -1,5 +1,35 @@
...
+                if (post.description.startsWith('#post')) {
```
Comment (tjefferson08) [inline apps/11gist-blog/script.js:9]: I like startsWith

File: apps/11gist-blog/script.js [hunk 14]
```diff
@@ -1,5 +1,35 @@
...
+                    var strPostDescription = post.description.slice(5);
+                    var $post = $('<li><a href="#" data-url="' + post.url + '" data-comments="' + post.comments_url + '">' + strPostDescription + '</a></li>');
```
Comment (tjefferson08) [inline apps/11gist-blog/script.js:11]: Might be easier to read if you broke onto multiple lines

File: apps/11gist-blog/script.js [hunk 15]
```diff
@@ -1,5 +1,35 @@
...
 'use strict';
 
 $(document).ready(function() {
-  // You code here
+    // You code here
+    $.ajax('https://api.github.com/users/jroman93/gists', {
+        success: function(posts) {
+            posts.forEach(function(post) {
+                if (post.description.startsWith('#post')) {
+                    var strPostDescription = post.description.slice(5);
```
Comment (tjefferson08) [inline apps/11gist-blog/script.js:10]: Good name

File: apps/11gist-blog/script.js [hunk 16]
```diff
@@ -1,5 +1,35 @@
...
+                    var $post = $('<li><a href="#" data-url="' + post.url + '" data-comments="' + post.comments_url + '">' + strPostDescription + '</a></li>');
+                    $('#posts').append($post);
+                };
+            })
+            $('body').on('click', 'a', function(event) {
```
Comment (tjefferson08) [inline apps/11gist-blog/script.js:15]: Interesting choice here. Can you think of another way to write this $().on call? This is slightly different from the approaches we've used in class

File: apps/11gist-blog/script.js [hunk 17]
```diff
@@ -1,5 +1,35 @@
+
 'use strict';
 
 $(document).ready(function() {
-  // You code here
+    // You code here
+    $.ajax('https://api.github.com/users/jroman93/gists', {
```
Comment (tjefferson08) [inline apps/11gist-blog/script.js:6]: woo real API

**Pull request closed without merging**